src/hdmf/build/map.py

```diff
@@ -1151,19 +1154,24 @@ def __flatten(self, sub_builder, subspec, manager):
         return tmp
 
     @docval({'name': 'builder', 'type': (DatasetBuilder, GroupBuilder),
-             'doc': 'the builder to construct the Container from'},
+             'doc': 'the builder to construct the AbstractContainer from'},
             {'name': 'manager', 'type': BuildManager, 'doc': 'the BuildManager for this build'},
-            {'name': 'parent', 'type': (Proxy, Container),
-             'doc': 'the parent Container/Proxy for the Container being built', 'default': None})
+            {'name': 'parent', 'type': (Proxy, AbstractContainer),
+             'doc': 'the parent AbstractContainer/Proxy for the AbstractContainer being built', 'default': None})
     def construct(self, **kwargs):
-        ''' Construct an Container from the given Builder '''
+        ''' Construct an AbstractContainer from the given Builder '''
         builder, manager, parent = getargs('builder', 'manager', 'parent', kwargs)
         cls = manager.get_cls(builder)
         # gather all subspecs
         subspecs = self.__get_subspec_values(builder, self.spec, manager)
         # get the constructor argument that each specification corresponds to
         const_args = dict()
+        if issubclass(cls, Data):
```

Comment:
Special treatment of the *data* constructor argument for *Data* container classes

Reply:
come to think of it, I should add a comment